### PR TITLE
Ensuring a property level deprecation warning is set for python codegen

### PR DESCRIPTION
**Pull request by stack72**
Related: #4741

### CODE REVIEW

**Comment by stack72** [comment]
@pgavlin this is what I am interested in you looking at - this is property level warnings for deprecations in Pulumi

This is the only way we get deprecation annotations in Python :/

File: pkg/codegen/python/gen.go [hunk 1]
```diff
@@ -294,6 +294,12 @@ func (mod *modContext) genAwaitableType(w io.Writer, obj *schema.ObjectType) str
 		fmt.Fprintf(w, "        if %s and not isinstance(%s, %s):\n", pname, pname, ptype)
 		fmt.Fprintf(w, "            raise TypeError(\"Expected argument '%s' to be a %s\")\n", pname, ptype)
 
+		if prop.DeprecationMessage != "" {
+			fmt.Fprintf(w, "        if %s is not None:\n", pname)
+			fmt.Fprintf(w, "            warnings.warn(\"%s\", DeprecationWarning)\n", prop.DeprecationMessage)
+			fmt.Fprintf(w, "            pulumi.log.warn(\"%s is deprecated: %s\")\n", pname, prop.DeprecationMessage)
```
Comment (lukehoban) [inline pkg/codegen/python/gen.go:300]: Curious - what is the UX of each of these?  And specifically, why do we need both?

Comment (stack72) [reply pkg/codegen/python/gen.go:300]: the pulumi will add to the diagnostics if the user hasn't setup a real python environment (e.g. I write Pulumi code, but I don't use auto-complete of pep etc when doing so so I don't get any Pythonisms in my IDE)

Comment (stack72) [reply pkg/codegen/python/gen.go:300]: Happy to remove what you feel is unnecessary though :)

Comment (lukehoban) [reply pkg/codegen/python/gen.go:300]: As long as they won't both show up together - this sounds fine.

**Pull request merged**